common/source/docs/common-network.rst

```diff
@@ -0,0 +1,141 @@
+- :ref:`NET_P1_PORT<NET_P1_PORT>`: This is the TCP or UDP port number.
+- :ref:`NET_P1_PROTOCOL<NET_P1_PROTOCOL>`: This registers this port with the ArduPilot Serial Manager as having a device attached which uses the specified protocol. This allows ArduPilot to use the data as if it had come from a UART attached device or connection.
+
+The figure below shows a typical single device setup for an Ethernet gimbal.
+
+.. image:: ../../../images/NET_Periph.png
+
+The figure below shows a typical vehicle system with multiple peripherals and an optional ground-based connection to a factory network for data retrieval and vehicle setup.
+
+.. image:: ../../../images/Net_Vehicle.png
+
+Server or Client?
+-----------------
+
+The choice of client or server is determined by the capabilities of the device being connected. If it is a TCP server, then ArduPilot's port should be a TCP Client. If its a UDP Client, then ArduPilot would be a UDP Server.
+
+In general if a device is a controllable object to the autopilot, it would be a server, always listening for commands (ESC, Gimbal,etc.). If a continuous data stream (like an airspeed sensor), it would probably be a client, able to send data without inquiry.
+
+Scripting
+=========
+
+Network access to LUA scripts is provided with several `socket style bindings <https://github.com/ArduPilot/ardupilot/blob/d90224280e416d0475eaadd06648bbf0daab7f7d/libraries/AP_Scripting/docs/docs/docs.lua>`__  (search for "socket") that allows the script to connect and manage UDP or TCP data flow to the network from the script.
+
+An example script implementing a simple web server application on the autopilot which allows access to the SD card (for rapid log downloading), and access to autopilot state(arming and GPS/AHRS location in this example) is available `here <https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Scripting/applets/net_webserver.lua>`__ and its `readme file here <https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Scripting/applets/net_webserver.md>`__
+
+Running the Webserver Script Under SITL
+---------------------------------------
+
+- run SITL
+- enable scripting (:ref:`SCR_ENABLE<SCR_ENABLE>` = 1), reboot SITL
+- set :ref:`SCR_VM_I_COUNT<SCR_VM_I_COUNT>` = 1000000, reboot SITL
+- connect local webrowser to "http://127.0.0.1:8080" (not types as shown to avoid https)
+
+Resulting web page
+
+.. image:: ../../../images/LUA_webserver.jpg
+
+Connecting to Webserver via Network Attached Autopilot
+------------------------------------------------------
+In order to connect an autopilot running the webserver to the local network to allow other PCs to access the server:
+
+- connect autopilot to the local network (directly or via Ethernet switch)
+- make sure ``NET_IPADDR`` is an address within the local network range or use the network DHCP server to set it
+- connect local webrowser to "http://<NET_IPADDR>:8080" (note: type exactly as shown... "https://" will not work)
+
```

Comment:
That's only one way to do it. I did this:
Another way to make the ArduPilot web server accessible to other PC's on a network or VPN, is to run a proxy on the a companion computer such as a Raspberry PI or Jetson Nano. This can be done by installing nginx and creating a site definition something like this
```
server {
	listen 80;
	server_name mycompanion;
	location / {
	proxy_pass http://<NET_IPADDR>:8080;
	}
}
```
where the proxy_pass url would be the address of the connected AP.

Reply:
a bit esoteric...many  other ways also....but do need to clarify line 130

Reply:
I guess - it just seemed to imply that this is the _only_ way - I wanted to make it clearer that there is at least one other way, but as you point out, perhaps many others.